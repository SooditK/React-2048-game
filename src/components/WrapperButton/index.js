import React from 'react';
import PropTypes from 'prop-types';
import styles from './wrapperButton.scss';
import Button from '../../components/Button';

// Add a wrapper css class for `Button`
export default function WrapperButton({ children, cls, onClick, ...props }) {
  return (
    <div className={`${styles.btn} ${cls}`}>
      {/* Pass down external props */}
      <Button onClick={onClick} {...props}>
        {children}
      </Button>
    </div>
  );
}

WrapperButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  cls: PropTypes.string
};

WrapperButton.defaultProps = {
  onClick() {},
  cls: ''
};
