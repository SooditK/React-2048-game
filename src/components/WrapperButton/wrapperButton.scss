@import '../../assets/styles/vars';

.btn {
  margin-right: 0.77vw;
  margin-top: 0.77vw;
}

@media all and (max-width: $break-point) {
  .btn {
    margin-right: 4vw;
    margin-top: 4vw;
  }
}
