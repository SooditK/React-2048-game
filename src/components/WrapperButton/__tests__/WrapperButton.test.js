import React from 'react';
import renderer from 'react-test-renderer';
import WrapperButton from '..';

describe('<WrapperButton />', () => {
  it('component render', () => {
    const btn = renderer
      .create(
        <WrapperButton>
          <p>test</p>
        </WrapperButton>
      )
      .toJSON();
    expect(btn).toMatchSnapshot();
  });
});
