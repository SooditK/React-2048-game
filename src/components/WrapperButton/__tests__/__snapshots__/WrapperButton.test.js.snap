// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrapperButton /> component render 1`] = `
<div
  className="btn "
>
  <button
    className="btn-flat btn-md btn-default"
    onClick={[Function]}
  >
    <p>
      test
    </p>
  </button>
</div>
`;
